ssl/ssl_lib.cc

```diff
@@ -3315,3 +3316,35 @@ int SSL_CTX_set1_curves_list(SSL_CTX *ctx, const char *curves) {
 int SSL_set1_curves_list(SSL *ssl, const char *curves) {
   return SSL_set1_groups_list(ssl, curves);
 }
+
+size_t SSL_client_hello_get0_ciphers(SSL *ssl, const unsigned char **out) {
+  STACK_OF(SSL_CIPHER) *client_cipher_suites = SSL_get_client_ciphers(ssl);
+  if (client_cipher_suites == nullptr) {
+      return 0;
+  }
+
+  size_t num_ciphers = sk_SSL_CIPHER_num(client_cipher_suites);
+  if (out != nullptr) {
+    // Hasn't been called before
+    if(ssl->client_cipher_suites_arr.empty()) {
+      if (!ssl->client_cipher_suites_arr.Init(num_ciphers)) {
+        OPENSSL_PUT_ERROR(SSL, ERR_R_MALLOC_FAILURE);
```

Comment:
You don't need to add this error here, Array.Init() calls OpenSSL_malloc which already adds an error if the system malloc fails. 
```suggestion
```

Reply:
`OPENSSL_malloc` [reports the error](https://github.com/aws/aws-lc/blob/637afcda8f495ade222e29e5e2e79ff770bf0211/crypto/mem.c#L201) against the `CRYPTO` library.
```
  OPENSSL_PUT_ERROR(CRYPTO, ERR_R_MALLOC_FAILURE);
```

There are [many other places](https://github.com/search?q=repo%3Aaws%2Faws-lc+path%3A%2F%5Essl%5C%2F%2F+ERR_R_MALLOC_FAILURE&type=code) that we also report this error for SSL.

Reply:
(Whoops, I deleted my previous comment by accident.)

We set the `ERR_R_MALLOC_FAILURE` for `SSL` in several other places as well: https://github.com/search?q=repo%3Aaws%2Faws-lc+path%3A%2Fssl%2F+ERR_R_MALLOC_FAILURE&type=code

Reply:
If malloc is already doing so, is there a specific reason for this redundancy? Maybe to ensure that the trace for the error stack is more detailed?

Reply:
Malloc reports the error against the "CRYPTO" module, not the "SSL" module as is done here. That seems to be the pattern but I'm. It sure how consistent it is.

Reply:
For clearer error logging, I'll leave this here

Reply:
TIL